service/device/registry/test-steps/src/main/java/org/eclipse/kapua/service/device/registry/steps/DeviceRegistrySteps.java

```diff
@@ -2559,4 +2560,65 @@ public void iFindDeviceWithClientId(String deviceName) {
 
         assertEquals(device.getClientId(), deviceName);
     }
+
+    @And("^I add device to group$")
+    public void iAddDeviceToGroup() throws Exception{
+        Device device = (Device) stepData.get("Device");
+        Group group = (Group) stepData.get("Group");
+
+        try {
+            KapuaId groupId = group.getId();
+            device.setGroupId(groupId);
+            Device newDevice = deviceRegistryService.update(device);
+            stepData.put("Device", newDevice);
+        } catch (Exception e) {
+            verifyException(e);
+        }
+    }
+
+    @When("^I change devices group to No Group$")
+    public void iChangeDevicesGroupToNoGroup() throws Exception{
+        Device device = (Device) stepData.get("Device");
+
+        try {
+            device.setGroupId(null);
+            device = deviceRegistryService.update(device);
+            stepData.put("Device", device);
+        } catch (Exception e) {
+            verifyException(e);
+        }
+    }
+
+    @And("^I create (\\d+) devices and add them to group$")
+    public void iCreateDevicesAndAddThemToGroup(int numberOfDevices) throws Exception {
```

Comment:
Change method implementation so it accepts group parameter: " public void iCreateDevicesAndAddThemToGroup(int numberOfDevices, String group) throws Exception {"